File: fixtures/extraction/ruby/test_roles/production_roles.rb
class ReportService
  def setup
    @rows = []
  end

  def teardown
    @rows = nil
  end

  def test_connection
    true
  end
end

describe "not a suite" do
end

before do
end

after do
end

test "not a rails case" do
end

let(:not_a_fixture) { 1 }

File: fixtures/extraction/swift/test_roles/production_roles.swift
import Testing

@Suite("Math suite")
struct MathSuite {
    let subject: Int

    init() {
        subject = 0
    }

    deinit {
        print("done")
    }

    @Test func addsTwoNumbers() {
        #expect(2 + 2 == 4)
    }

    @Test("adds many", arguments: [1, 2, 3])
    func addsManyNumbers(value: Int) {
        #expect(value > 0)
    }

    func makeSubject() -> Int {
        subject
    }
}

@Test func addsAtTopLevel() {
    #expect(1 == 1)
}

final class NetworkClient {
    init() { }

    deinit { }

    func testConnection() -> Bool {
        true
    }

    func setUp() { }
}
